test/commands/InspectCommandTest.ts

```diff
@@ -0,0 +1,181 @@
+import {BaseMocks} from "@lambocreeper/mock-discord.js";
+
+import InspectCommand from "../../src/commands/InspectCommand";
+import Command from "../../src/abstracts/Command";
+import {EMBED_COLOURS} from "../../src/config.json";
+import DateUtils from "../../src/utils/DateUtils";
+import exp = require("constants");
+
+describe("InspectCommand", () => {
+	describe("constructor()", () => {
+		it("creates a command called inspect", () => {
+			const command = new InspectCommand();
+
+			expect(command.getName()).to.equal("inspect");
+		});
+
+		it("creates a command with correct description", () => {
+			const command = new InspectCommand();
+
+			expect(command.getDescription()).to.equal("Show information about a given user");
+		});
+	});
+
+	describe("run()", () => {
+		let sandbox: SinonSandbox;
+		let message: Message;
+		let command: Command;
+
+		beforeEach(() => {
+			sandbox = createSandbox();
+			command = new InspectCommand();
+			message = BaseMocks.getMessage();
+		});
+
+		it("sends a message to the channel", async () => {
+			const messageMock = sandbox.stub(message.channel, "send");
+
+			await command.run(message, ["User"]);
+
+			expect(messageMock.calledOnce).to.be.true;
+		});
+
+		it("sends an error message when user is not found", async () => {
+			const messageMock = sandbox.stub(message.channel, "send");
+
+			sandbox.stub(GuildMemberManager.prototype, "fetch").resolves(new Collection([]));
+
+			await command.run(message, ["FakeUser#1234"]);
+
+			const embed = messageMock.getCall(0).firstArg.embed;
+
+			expect(messageMock.calledOnce).to.be.true;
+			expect(embed.title).to.equal("Error");
+			expect(embed.description).to.equal("No user was found with this username/userID.");
+			expect(embed.hexColor).to.equal(EMBED_COLOURS.ERROR.toLowerCase());
+		});
+
+		it("sends an error message username has incorrect format", async () => {
+			const messageMock = sandbox.stub(message.channel, "send");
+
+			sandbox.stub(GuildMemberManager.prototype, "fetch").resolves(new Collection([]));
+
+			await command.run(message, ["FakeUser#123"]);
+
+			const embed = messageMock.getCall(0).firstArg.embed;
+
+			expect(messageMock.calledOnce).to.be.true;
+			expect(embed.title).to.equal("Error");
+			expect(embed.description).to.equal("Incorrect usage of command");
+			expect(embed.fields[0].name).to.equal("Correct Usage");
+			expect(embed.fields[0].value).to.equal("?inspect [username + discriminator / userID]");
+			expect(embed.hexColor).to.equal(EMBED_COLOURS.ERROR.toLowerCase());
+		});
+
+		it("sends a message with information if the argument was a username", async () => {
+			const messageMock = sandbox.stub(message.channel, "send");
+			const member = BaseMocks.getGuildMember();
+
+			sandbox.stub(GuildMemberManager.prototype, "fetch").resolves(new Collection([["12345", member]]));
+
+			sandbox.stub(GuildMemberRoleManager.prototype, "cache").get(() => new Collection([["12345", new Role(BaseMocks.getClient(), {
```

Comment:
You define this exact same role 3 times, maybe you can put it in a variable to not repeat yourself?